parsers/yacc.c

```diff
@@ -52,12 +54,20 @@ static  void change_section (const char *line CTAGS_ATTR_UNUSED,
 		c_source_start = getSourceLineNumber();
 
 		/* Skip the lines for finding the EOF. */
-		while (readLineFromInputFile ())
-			;
+		endCharOffset = 0;
+		last = NULL;
+		while ((tmp = readLineFromInputFile ()))
+			last = tmp;
+		if (last)
+			endCharOffset = strlen ((const char *)last);
+		/* If `last' is too long, strlen returns a too large value
+		   for the positive area of `endCharOffset'. */
```

Comment:
Oh.  OK well no then I guess it's fine, disregard that comment.  And yeah that FILE API isn't the best ever, but well.